fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/group/ChunkReader.java

```diff
@@ -0,0 +1,328 @@
+                    expandGroup2Patients(fhirTenant, fhirDatastoreId, group2, patients, groupsInPath);
+                }
+            }
+        }
+    }
+
+    private Group findGroupByID(String fhirTenant, String fhirDatastoreId, String groupId) throws Exception{
+        FHIRRequestContext.set(new FHIRRequestContext(fhirTenant, fhirDatastoreId));
+        FHIRPersistenceHelper fhirPersistenceHelper = new FHIRPersistenceHelper();
+        fhirPersistence = fhirPersistenceHelper.getFHIRPersistenceImplementation();
+
+        FHIRSearchContext searchContext;
+        FHIRPersistenceContext persistenceContext;
+        Map<String, List<String>> queryParameters = new HashMap<>();
+
+        queryParameters.put("_id", Arrays.asList(new String[] { groupId }));
+        searchContext = SearchUtil.parseQueryParameters(Group.class, queryParameters);
+        List<Resource> resources = null;
+        FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+        txn.begin();
+        persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+        resources = fhirPersistence.search(persistenceContext, Group.class).getResource();
+        txn.commit();
+
+        if (resources != null) {
+            return (Group) resources.get(0);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public Object readItem() throws Exception {
+        List <String> allCompartmentResourceTypes = CompartmentUtil.getCompartmentResourceTypes("Patient");
+        if (fhirResourceType == null ) {
+            resourceTypes = allCompartmentResourceTypes;
+        } else {
+            List<String> tmpResourceTypes = Arrays.asList(fhirResourceType.split("\\s*,\\s*"));
+            resourceTypes = tmpResourceTypes.stream().filter(item-> allCompartmentResourceTypes.contains(item)).collect(Collectors.toList());
+            if (resourceTypes == null || resourceTypes.isEmpty()) {
+                throw new Exception("readItem: None of the input resource types is valid!");
+            }
+        }
+
+        if (fhirSearchPatientGroupId == null) {
+            throw new Exception("readItem: missing group id for this group export job!");
+        }
+
+        TransientUserData chunkData = (TransientUserData) jobContext.getTransientUserData();
+        if (chunkData != null && pageNum > chunkData.getLastPageNum()) {
+            if (resourceTypes.size() == indexOfCurrentResourceType + 1) {
+                // No more resource type and page to read, so return null to end the reading.
+                return null;
+            } else {
+                // More resource types to read, so reset pageNum, partNum and move resource type index to the next.
+                pageNum = 1;
+                chunkData.setPartNum(1);
+                indexOfCurrentResourceType++;
+            }
+        }
+        if (fhirTenant == null) {
```

Comment:
I think we should throw an exception.  This could lead to unintended consequences.  if a tenant is not passed in, it should be treated as invalid.

Reply:
the control should be in the operation side, the javabatch itself allows non tenant id to make the javabatch Job test itself easier.